source/slang/core.meta.slang

```diff
@@ -2650,6 +2651,9 @@ attribute_syntax [__vulkanHitAttributes] : VulkanHitAttributesAttribute;
 __attributeTarget(FunctionDeclBase)
 attribute_syntax [mutating] : MutatingAttribute;
 
+__attributeTarget(FunctionDeclBase)
+attribute_syntax [semanticmutating] : SemanticMutatingAttribute;
```

Comment:
`semanticmutating` without spacing, underscore and camel-casing is little hard to read.

If this is going to be used only internally, it should start with double underscore like `[__semanticmutating]`.
If it will be used by the users, we should put this in the document and explain how it works with an example.

Reply:
Good point, I think for now making it an internal feature only is enough, since it was necessary to only fix warnings in the stdlib.